modules/runner/src/test/java/org/apache/ignite/internal/configuration/storage/DistributedConfigurationStorageTest.java

```diff
@@ -74,6 +93,145 @@ void stop() throws Exception {
         vaultManager.stop();
     }
 
+    /**
+     * Dummy configuration.
+     */
+    @ConfigurationRoot(rootName = "someKey", type = DISTRIBUTED)
+    public static class DistributedTestConfigurationSchema {
+        @Value(hasDefault = true)
+        public final int foobar = 0;
+    }
+
+    /**
+     * Tests that distributed configuration storage correctly picks up latest configuration MetaStorage revision
+     * during recovery process.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testMetaStorageRevisionDifferentFromConfigurationOnRestart() throws Exception {
+        RootKey<DistributedTestConfiguration, DistributedTestView> rootKey = DistributedTestConfiguration.KEY;
+
+        ConfigurationAsmGenerator cgen = new ConfigurationAsmGenerator();
+
+        MetaStorageMockWrapper wrapper = new MetaStorageMockWrapper();
+
+        int configurationChangesCount = 7;
```

Comment:
why 7? Isn't one change enough?

Reply:
It's a tribute to the original bug I was constantly facing. It's extremely fast, so I feel like it's ok? Or I can change it if it looks cryptic

Reply:
It looks cryptic indeed, can you add a comment?

Reply:
Sure